client/my-sites/plugins/main.jsx

```diff
@@ -142,8 +142,10 @@ export default React.createClass( {
 	},
 
 	getPluginsState( nextProps ) {
-		const sites = this.state && this.state.bulkManagement ? this.props.sites.getSelectedOrAllWithPlugins() : this.props.sites.getSelectedOrAll(),
-			pluginUpdate = PluginsStore.getPlugins( sites, 'updates' );
+		const sites = this.state && this.state.bulkManagement.length
+			? this.props.sites.getSelectedOrAllWithPlugins()
+			: this.props.sites.getSelectedOrAll();
```

Comment:
JavaScript is an *eagerly-evaluated language*, meaning that regardless of the length of `bulkManagement` here, we end up calling both of these functions. If we only wanted to run one, we could create an intermediate function variable to make this lazy, meaning we only call the appropriate function according to the ternary expression.

```js
const { bulkManagement = [] } = this.state;
const { getSelectedOrAllWithPlugins, getSelectedOrAll } = this.props.sites;

const siteGetter = bulkManagement.length ? getSelectedOrAllWithPlugins : getSelectedOrAll;
const sites = siteGetter();

// or via an [IIFE](https://en.wikipedia.org/wiki/Immediately-invoked_function_expression)
const sites = ( bulkManagement.length ? getSelectedOrAllWithPlugins : getSelectedOrAll )();
```